algebird-core/src/main/scala/com/twitter/algebird/BloomFilter.scala

```diff
@@ -225,6 +236,12 @@ case class BFInstance(hashes: BFHash, bits: BitSet, width: Int) extends BF {
       width)
   }
 
+  def checkAndAdd(item: String): (BF, ApproximateBoolean) = {
+    val bitsToActivate = BitSet(hashes(item): _*)
+    val contained = this.contains(item)
+    (BFInstance(hashes, bits ++ bitsToActivate, width), contained)
+  }
+
   def bitSetContains(bs: BitSet, il: Int*): Boolean = {
     il.map{ i: Int => bs.contains(i) }.reduce{ _ && _ }
```

Comment:
Defintiely. I considered this ealier - performance measurements showed hash calculations and bit array conversions trumped everything small like this. If we do add this, how about:     
```
il.foreach { i => if (!bs.contains(i)) return false }
    return true
```
Not sure how we feel about break-out returns on Scala, but we will avoid extra assignments.